ElectronClient/gui/KeymapConfig/KeymapConfigScreen.tsx

```diff
@@ -0,0 +1,191 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { KeymapItem } from '../../lib/services/KeymapService';
+import { ShortcutRecorder } from './ShortcutRecorder';
+import getLabel from './utils/getLabel';
+import useKeymap from './utils/useKeymap';
+import useCommandStatus from './utils/useCommandStatus';
+import styles_ from './styles';
+
+const { bridge } = require('electron').remote.require('./bridge');
+const { shim } = require('lib/shim');
+const { _ } = require('lib/locale');
+
+export interface KeymapConfigScreenProps {
+	themeId: number
+}
+
+export const KeymapConfigScreen = ({ themeId }: KeymapConfigScreenProps) => {
+	const styles = styles_(themeId);
+
+	const [filter, setFilter] = useState('');
+	const [keymapItems, keymapError, overrideKeymapItems, exportCustomKeymap, setAccelerator, resetAccelerator] = useKeymap();
+	const [recorderError, setRecorderError] = useState<Error>(null);
+	const [editing, enableEditing, disableEditing] = useCommandStatus();
+	const [hovering, enableHovering, disableHovering] = useCommandStatus();
+
+	const handleSave = (event: { commandName: string, accelerator: string }) => {
+		const { commandName, accelerator } = event;
+		setAccelerator(commandName, accelerator);
+		disableEditing(commandName);
+	};
+
+	const handleReset = (event: { commandName: string }) => {
+		const { commandName } = event;
+		resetAccelerator(commandName);
+		disableEditing(commandName);
+	};
+
+	const handleCancel = (event: { commandName: string }) => {
+		const { commandName } = event;
+		disableEditing(commandName);
+	};
+
+	const handleError = (event: { recorderError: Error }) => {
+		const { recorderError } = event;
+		setRecorderError(recorderError);
+	};
+
+	const handleImport = async () => {
+		const filePath = bridge().showOpenDialog({
+			properties: ['openFile'],
+			defaultPath: 'keymap-desktop',
+			filters: [{ name: 'Joplin Keymaps (keymap-desktop.json)', extensions: ['json'] }],
+		});
+
+		if (filePath) {
+			const actualFilePath = filePath[0];
+			try {
+				const keymapFile = await shim.fsDriver().readFile(actualFilePath, 'utf-8');
+				overrideKeymapItems(JSON.parse(keymapFile));
+			} catch (err) {
+				bridge().showErrorMessageBox(`${_('An unexpected error occured while importing the keymap!')}\n${err.message}`);
+			}
+		}
+	};
+
+	const handleExport = async () => {
+		const filePath = bridge().showSaveDialog({
+			defaultPath: 'keymap-desktop',
+			filters: [{ name: 'Joplin Keymaps (keymap-desktop.json)', extensions: ['json'] }],
+		});
+
+		if (filePath) {
+			try {
+				// KeymapService is already synchronized with the in-state keymap
+				await exportCustomKeymap(filePath);
+			} catch (err) {
+				bridge().showErrorMessageBox(`${_('An unexpected error occured while exporting the keymap!')}\n${err.message}`);
```

Comment:
As an error here is unlikely, let's not create a new translatable string for it. So just `bridge().showErrorMessageBox(err.message);` for now would work.

Reply:
Sure, I'll remove it right away.